dsgrid/registry/dimension_mapping_registry_manager.py

```diff
@@ -103,6 +107,27 @@ def check_unique_records(self, config: DimensionMappingsConfig, warn_only=False)
                     f"There are {len(duplicates)} dimension mappings with duplicate content (data files)."
                 )
 
+    def _check_records_against_dimension_records(self, config):
+        for mapping in config.model.mappings:
+            actual_from_records = {x.from_id for x in mapping.records}
+            actual_to_records = {x.to_id for x in mapping.records}
+            from_dimension = self._dimension_mgr.get_by_id(mapping.from_dimension.dimension_id)
+            allowed_from_records = from_dimension.get_unique_ids()
+            diff = actual_from_records.difference(allowed_from_records)
```

Comment:
Do we only care the excess difference from `actual` to `allowed`? What about when `actual` is smaller than `allowed`, (i.e. A.symmetric_difference(B))?

Same goes for the difference for `to_dimension` below.

Reply:
I think actual can be smaller than allowed, but I wonder if we want the mapping (in its config) to specify that it represents a subset if that's the case?

Reply:
My takeaway from today's meeting was that a dimension mapping record file must include ALL records from the dimension (Lixi's solution). Implemented.

Reply:
Implemented per our email discussion. Since dimension mappings are not tied to dimension associations, it can cause problems for users.

Reply:
An alternative that I'm sure no one will like is to require the dataset contributor to add `from=null, to=X` for all records that do not belong to their data source.